kmcd-data.md

```diff
@@ -0,0 +1,294 @@
+                 | Rad "^Rad" Int [function]
+                 > Rad "+Rad" Rad [function]
+                 | Rad "-Rad" Rad [function]
+ // ----------------------------------------
+    rule R1 *Rad R2 => R1 *Rat R2
+    rule R1 /Rad R2 => R1 /Rat R2
+    rule R1 ^Rad R2 => R1 ^Rat R2
+    rule R1 +Rad R2 => R1 +Rat R2
+    rule R1 -Rad R2 => R1 -Rat R2
+```
+
+```k
+    syntax Wad ::= minWad ( Wad , Wad ) [function]
+                 | maxWad ( Wad , Wad ) [function]
+ // ----------------------------------------------
+    rule minWad(W1, W2) => minRat(W1, W2)
+    rule maxWad(W1, W2) => maxRat(W1, W2)
+
+    syntax Ray ::= minRay ( Ray , Ray ) [function]
+                 | maxRay ( Ray , Ray ) [function]
+ // ----------------------------------------------
+    rule minRay(W1, W2) => minRat(W1, W2)
+    rule maxRay(W1, W2) => maxRat(W1, W2)
+
+    syntax Rad ::= minRad ( Rad , Rad ) [function]
+                 | maxRad ( Rad , Rad ) [function]
+ // ----------------------------------------------
+    rule minRad(W1, W2) => minRat(W1, W2)
+    rule maxRad(W1, W2) => maxRat(W1, W2)
+```
+
+```k
+    syntax Bool ::= Wad  "<=Wad" Wad [function]
+                  | Wad   "<Wad" Wad [function]
+                  | Wad  ">=Wad" Wad [function]
+                  | Wad   ">Wad" Wad [function]
+                  | Wad  "==Wad" Wad [function]
+                  | Wad "=/=Wad" Wad [function]
+ // -------------------------------------------
+    rule W1  <=Wad W2 => W1  <=Rat W2
+    rule W1   <Wad W2 => W1   <Rat W2
+    rule W1  >=Wad W2 => W1  >=Rat W2
+    rule W1   >Wad W2 => W1   >Rat W2
+    rule W1  ==Wad W2 => W1  ==Rat W2
+    rule W1 =/=Wad W2 => W1 =/=Rat W2
+
+    syntax Bool ::= Ray  "<=Ray" Ray [function]
+                  | Ray   "<Ray" Ray [function]
+                  | Ray  ">=Ray" Ray [function]
+                  | Ray   ">Ray" Ray [function]
+                  | Ray  "==Ray" Ray [function]
+                  | Ray "=/=Ray" Ray [function]
+ // -------------------------------------------
+    rule W1  <=Ray W2 => W1  <=Rat W2
+    rule W1   <Ray W2 => W1   <Rat W2
+    rule W1  >=Ray W2 => W1  >=Rat W2
+    rule W1   >Ray W2 => W1   >Rat W2
+    rule W1  ==Ray W2 => W1  ==Rat W2
+    rule W1 =/=Ray W2 => W1 =/=Rat W2
+
+    syntax Bool ::= Rad  "<=Rad" Rad [function]
+                  | Rad   "<Rad" Rad [function]
+                  | Rad  ">=Rad" Rad [function]
+                  | Rad   ">Rad" Rad [function]
+                  | Rad  "==Rad" Rad [function]
+                  | Rad "=/=Rad" Rad [function]
+ // -------------------------------------------
+    rule W1  <=Rad W2 => W1  <=Rat W2
+    rule W1   <Rad W2 => W1   <Rat W2
+    rule W1  >=Rad W2 => W1  >=Rat W2
+    rule W1   >Rad W2 => W1   >Rat W2
+    rule W1  ==Rad W2 => W1  ==Rat W2
+    rule W1 =/=Rad W2 => W1 =/=Rat W2
+```
+
+```k
+    syntax Rad ::= Wad "*RateWad" Ray [function]
+                 | Rad "*RateRad" Ray [function]
```

Comment:
`*RateRad` doesn't seem to be used anywhere, and it really shouldn't be necessary...in that case, we can get rid of it and `*RateWad` can just be `*Rate`.

Reply:
Although IIRC, you said there was some reason for this on the call...but I've managed to forget it between this morning and tonight.

But `*RateRad` wouldn't result in a `Rad` anyway...it would result in a (45+27) digit fixed point number. What you really want, I think, is the `rmul(Rad, Ray)` I mentioned earlier. If `K` really won't do parametric typing in a nice way (and since you aren't already using it, I suspect it doesn't), we can have to operators:
`syntax Wad ::= wrmul(Wad, Ray)` (or call it `*WadRay`)
`syntax Rad ::= rrmul(Rad, Ray)` (or call it `*RadRay`)

Reply:
Ah it was hidden in the kmcd-props.md diff (not loaded by default)

Reply:
Looking at how it's being used, some sort of `rmul` is definitely the right way to express this.

Reply:
Same here, I've switch to `rmulRad`. I can't just have it be `rmul` until we actually switch over to separate representations for each type (next PR), because the sort inferencer will get confused.